configure.ac

```diff
@@ -241,38 +241,44 @@ AC_SEARCH_LIBS([clock_gettime], [rt],
 dnl check for libraries
 
 OMPI_CHECK_PORTALS4([portals4],
-    [AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
-     transport="portals4"
+    [transport="portals4"
      transport_portals4="yes"],
     [transport_portals4="no"])
-AM_CONDITIONAL([USE_PORTALS4], [test "$transport_portals4" = "yes"])
 
-OPAL_CHECK_LIBFABRIC([libfabric],
-    [AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
-     transport="ofi"
+OPAL_CHECK_OFI([ofi],
+    [transport="ofi"
      transport_ofi="yes"],
     [transport_ofi="no"])
-AM_CONDITIONAL([USE_OFI], [test "$transport_ofi" = "yes"])
 
-if test "$transport_ofi" = "no" ; then
-  OMPI_CHECK_OFI([ofi],
-      [AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
-       transport="ofi"
-       transport_ofi="yes"],
-      [transport_ofi="no"])
-  AM_CONDITIONAL([USE_OFI], [test "$transport_ofi" = "yes"])
+# assure that only ofi is enabled if portals4 is also detected
+if test -n "$with_ofi" ; then
+    transport="ofi"
+    transport_portals4="no"
+    AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
+elif test -n "$with_portals4" ; then
+    transport="portals4"
+    transport_ofi="no"
+    AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
 fi
+AM_CONDITIONAL([USE_OFI], [test "$transport_ofi" = "yes"])
+AM_CONDITIONAL([USE_PORTALS4], [test "$transport_portals4" = "yes"])
 
 SANDIA_CHECK_XPMEM(
-    [AC_DEFINE([USE_XPMEM], [1], [Define if XPMEM transport is active])
-     transport_xpmem="yes"],
+    [transport_xpmem="yes"],
     [transport_xpmem="no"])
-AM_CONDITIONAL([USE_XPMEM], [test "$transport_xpmem" = "yes"])
 
 CHECK_CMA(
-   [AC_DEFINE([USE_CMA], [1], [Define if Cross Memory Attach transport is active])
-    transport_cma="yes"],
-   [transport_cma="no"])
+    [transport_cma="yes"],
+    [transport_cma="no"])
+
+if test -n "$with_xpmem" ; then
+    transport_cma="no"
+    AC_DEFINE([USE_XPMEM], [1], [Define if XPMEM transport is active])
```

Comment:
On second thought, I think $with_X is a poor choice here, because I think it's *possible* for $with_X to equal "no" in most, if not all of these scripts.  Also, we should try to avoid introducing variables that are defined in m4 scripts outside of configure.ac.  Does my latest commit fix this?